site/src/pages/WorkspacePage/WorkspaceNotifications.tsx

```diff
@@ -0,0 +1,343 @@
+import { workspaceResolveAutostart } from "api/queries/workspaceQuota";
+import { Template, TemplateVersion, Workspace } from "api/typesGenerated";
+import { AlertProps } from "components/Alert/Alert";
+import { FC, ReactNode, useEffect, useState } from "react";
+import { useQuery } from "react-query";
+import { WorkspacePermissions } from "./permissions";
+import dayjs from "dayjs";
+import { useIsWorkspaceActionsEnabled } from "components/Dashboard/DashboardProvider";
+import formatDistanceToNow from "date-fns/formatDistanceToNow";
+import { Pill } from "components/Pill/Pill";
+import InfoOutlined from "@mui/icons-material/InfoOutlined";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "components/Popover/Popover";
+import { Interpolation, Theme, useTheme } from "@emotion/react";
+import Button, { ButtonProps } from "@mui/material/Button";
+import { ThemeRole } from "theme/experimental";
+import WarningRounded from "@mui/icons-material/WarningRounded";
+
+type Notification = {
+  title: string;
+  severity: AlertProps["severity"];
+  detail?: ReactNode;
+  actions?: ReactNode;
+};
+
+type WorkspaceNotificationsProps = {
+  workspace: Workspace;
+  template: Template;
+  permissions: WorkspacePermissions;
+  onRestartWorkspace: () => void;
+  onUpdateWorkspace: () => void;
+  onActivateWorkspace: () => void;
+  latestVersion?: TemplateVersion;
+};
+
+export const WorkspaceNotifications: FC<WorkspaceNotificationsProps> = (
+  props,
+) => {
+  const {
+    workspace,
+    template,
+    latestVersion,
+    permissions,
+    onRestartWorkspace,
+    onUpdateWorkspace,
+    onActivateWorkspace,
+  } = props;
```

Comment:
```suggestion
export const WorkspaceNotifications: FC<WorkspaceNotificationsProps> = ({
  workspace,
  template,
  latestVersion,
  permissions,
  onRestartWorkspace,
  onUpdateWorkspace,
  onActivateWorkspace,
}) => {
```

I think we opted to keep inline prop destructuring, as long as we're using the `FC` type